triton/apps/llms.rst

```diff
@@ -0,0 +1,220 @@
+  :widths: 1 1 3 2
+
+  * * Model type
+    * Model version
+    * Module command to load
+    * Description
+
+  * * Llama 2
+    * 
+    * Module command to load
+    * Description
+
+All Huggingface models can be loaded with:  ``module load model-huggingface/all``,
+Here is a python script using huggingface model.
+
+.. code-block:: python
+
+  #force transformer to use local hub instead of download from remote hub
+  import os
+  os.environ['TRANSFORMERS_OFFLINE'] = '1'
+
+  from transformers import AutoModelForCausalLM, AutoTokenizer
+
+  tokenizer = AutoTokenizer.from_pretrained("mistralai/Mistral-7B-v0.1")
+  model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.1")
+  prompt = "How many stars in the space?"
+  model_inputs = tokenizer([prompt], return_tensors="pt")
+  input_length = model_inputs.input_ids.shape[1]
+  generated_ids = model.generate(**model_inputs, max_new_tokens=20)
+  print(tokenizer.batch_decode(generated_ids[:, input_length:], skip_special_tokens=True)[0])
+
+
+
+llama.cpp and GGUF
+~~~~~~~~~~~~~~~~~~~
+
+`llama.cpp <https://github.com/ggerganov/llama.cpp>`__ is a popular framework
+for running inference on LLM models with CPUs or GPUs. llama.cpp uses a format
+called GGUF as its storage format.
+
+We have llama.cpp conversions of all models with multiple quantizations levels.
+
+Before loading the modules load a module for the model weight you want to use.
+
+.. list-table::
+  :header-rows: 1
+  :widths: 1 1 3 2
+
+  * * Model type
+    * Model version
+    * Module command to load
+    * Description
+
+  * * Llama 2
+    * f16-2023-08-28
+    * ``module load model-llama.cpp/f16-2023-08-28`` (after loading a Llama 2 model for some weight)
+    * Half precision version of Llama 2 weights done with llama.cpp on 28th of Aug 2023.
+
+  * * Llama 2
+    * q4_0-2023-08-28
+    * ``module load model-llama.cpp/q4_0-2023-08-28`` (after loading a Llama 2 model for some weight)
+    * 4-bit integer version of Llama 2 weights done with llama.cpp on 28th of Aug 2023.
+
+  * * Llama 2
+    * q4_1-2023-08-28
+    * ``module load model-llama.cpp/q4_1-2023-08-28`` (after loading a Llama 2 model for some weight)
+    * 4-bit integer version of Llama 2 weights done with llama.cpp on 28th of Aug 2023.
+
+  * * Llama 2
+    * q8_0-2023-08-28
+    * ``module load model-llama.cpp/q8_0-2023-08-28`` (after loading a Llama 2 model for some weight)
+    * 8-bit integer version of Llama 2 weights done with llama.cpp on 28th of Aug 2023.
+
+Each module will set the following environment variables:
+
+- ``MODEL_ROOT`` - Folder where model weights are stored.
+- ``MODEL_WEIGHTS`` - Path to the model weights in GGUF format.
+
+This Python code snippet is part of a 'Chat with Your PDF Documents' example, utilizing LangChain and leveraging model weights stored in a .gguf file. For detailed environment setting up and python code, please check out `this repo <https://github.com/AaltoSciComp/llm-examples/tree/main/chat-with-pdf>`__.
```

Comment:
llm-examples is still private (I guess this will be fixed before merging)

Reply:
Also please fix the python code in chat-with-pdf to accept pdf as a CLI argument.
Mentioning how to exit the chat interface is also useful.